tests/test_gradient.py

```diff
@@ -49,7 +38,7 @@ def test_gradient_checkpointing(self, shape, kernel, space_order):
 
         gradient, _ = wave.jacobian_adjoint(residual, u0, vp=v0, checkpointing=True)
         gradient2, _ = wave.jacobian_adjoint(residual, u0, vp=v0, checkpointing=False)
-        assert np.allclose(gradient.data, gradient2.data)
+        assert np.allclose(gradient.data, gradient2.data, atol=0, rtol=0)
```

Comment:
interesting, what's the difference without these additions?

Reply:
`rtol=1e-05, atol=1e-08` that's the numpy defaults

Reply:
In effect you are asking if they are bitwise identical.

Reply:
so equivalent to `np.all(gradient.data == gradient2.data)` ?

Reply:
Yes.